src/main/java/net/sacredlabyrinth/phaed/simpleclans/commands/BanCommand.java

```diff
@@ -38,22 +39,28 @@ public void execute(Player player, String[] arg) {
 
 		UUID uuid = UUIDMigration.getForcedPlayerUUID(arg[0]);
 		if (uuid == null) {
-            ChatBlock.sendMessage(player, ChatColor.RED + plugin.getLang("no.player.matched"));
-            return;
-        }
+            		ChatBlock.sendMessage(player, ChatColor.RED + plugin.getLang("no.player.matched"));
+            		return;
+        	}
 		
 		if (plugin.getSettingsManager().isBanned(uuid)) {
 			ChatBlock.sendMessage(player, ChatColor.RED + plugin.getLang("this.player.is.already.banned"));
 			return;
 		}
+	    
+	        ClanPlayer cp = plugin.getClanManager().getClanPlayer(uuid);
 
-		Player pl = SimpleClans.getInstance().getServer().getPlayer(uuid);
```

Comment:
The actual problem is that if you ban a player from clan, then unban, and then you try to ban again, you get this NullPointerException.

That line I just moved to be under the ban method

Reply:
This was already present:
```
	        Player pl = SimpleClans.getInstance().getServer().getPlayer(uuid);
	    	if (pl != null) {
			ChatBlock.sendMessage(pl, ChatColor.AQUA + plugin.getLang("you.banned"));
		}
```

I just moved it below, because it was sending the message to the player and he wasnt banned because of the NullPointerException

Reply:
There is a simpler way to fix this NPE:
https://github.com/marcelo-mason/SimpleClans/blob/a7cb9ed1712dcc1c9a79f25cacef4d1d33386afa/src/main/java/net/sacredlabyrinth/phaed/simpleclans/managers/ClanManager.java#L552
Checking if cp is null here.

Reply:
Oh yeah it can be done like that, but as both required a clanplayer I just created a ban method which receives a clan player.

The only problem of fixing this NPE that way is that the command sender wont get feedback, so it will try to ban someone and nothing would happen (unless you add a checker like I did in the command). But as in the command I got the ClanPlayer, I just created the method so we wont getClanPlayer twice

Reply:
> nothing would happen

The player will be banned.

Reply:
Okay, now I added it as it was before but I added the checker. I now understand what you were trying to tell me, sorry!

Reply:
Thanks!